pymeasure/experiment/parameters.py

```
@@ -498,7 +562,137 @@ def __repr__(self):
+    def serialize(self):
+        """Act on the parameter values and specify what the group should return.
+           By default it returns a list of the parameter values.
+           Modify in subclasses.
+        """
+        return [parameter.value for (_, parameter) in self.parameters.items()]
+
+    def deserialize(self, value: Union[list, str]):
+        if isinstance(value, str):
+            value = json.loads(value)
+            value = [item for _,item in value.items()]
+        for value, (_, parameter) in zip(value, self.parameters.items()):
```

Ruff (0.15.10)

[warning] Loop control variable `value` overrides iterable it iterates

(B020)